pynet/datasets/lpsnc.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# NSAp - Copyright (C) CEA, 2019
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+# @Author: Zhou Kai
+# @GitHub: https://github.com/athon2
+# @Date: 2018-11-30 09:53:44
+##########################################################################
+
+"""
+Module that provides functions to prepare the lpsnc dataset.
+"""
+
+# Imports
+from collections import namedtuple
+import os
+import logging
+import numpy as np
+import pandas as pd
+import nibabel as nib
+import progressbar
+from pynet.datasets import Fetchers
+
+# Global parameters
+MODALITIES = ('T2w', 'T1w', 'ce-GADOLINIUM_T1w', 'FLAIR')
+Item = namedtuple("Item", ["input_path", "output_path", "metadata_path"])
+logger = logging.getLogger("pynet")
+
+
+# datasetdir="/neurospin/radiomics/workspace_LPSNC/LPSNC_data"
+@Fetchers.register
+def fetch_lpsnc(datasetdir, modality=0):
+    """ Fetch/prepare the lpsnc dataset for pynet.
+
+    Parameters
+    ----------
+    datasetdir: str
+        the dataset destination folder.
+
+    Returns
+    -------
+    item: namedtuple
+        a named tuple containing 'input_path', 'output_path', and
+        'metadata_path'.
+    """
+    if modality not in MODALITIES and modality != 0:
+        raise ValueError(
+                "Expect modality==0 for all modalities "
+                + "or modality in ('T2w','T1w','ce-GADOLINIUM_T1w','FLAIR')")
+    logger.info("Loading lpsnc dataset.")
+
+    def _crop(arr):
+        # return arr[45: 195, 30: 220, 10: 145]
+        return arr
+
+    def _norm(arr):
+        arr = np.nan_to_num(arr)
+        logical_mask = (arr != 0)
+        mean = arr[logical_mask].mean()
+        std = arr[logical_mask].std()
+        return ((arr - mean) / std).astype(np.single)
+
+    def merge_rois(roifiles, v=1):
+        """ Merge the input ROIs.
+
+        Parameters
+        ----------
+        roifiles: list of str
+            the ROI images to be merged.
+        outdir: str
+            the destination folder.
+        fmaskname: str
```

Comment:
fmaskname and outdir are not defined as input parameters.
Add docstring for the the v parameter, and use a more explicit parameter name for instance mask_value.